### PR TITLE
Added ViewBinding support, refractored some files and updated dependencies

### PR DESCRIPTION
**Pull request by MadRatSRP**

### CODE REVIEW

**Comment by esreli** [comment]
Thank you for your contribution @MadRatSRP. When some time frees up, our team will review your PR; this should be fairly soon. Thanks!

**Comment by MadRatSRP** [comment]
No problem, have a nice day :)

**Review by puneet-pdx — CHANGES_REQUESTED** [review]
@MadRatSRP Thank you for your contribution. 

Overall PR looks good. I have left some comments regarding some reformatting and refactoring. 
Also since this PR cannot be directly merged into the `main` branch. I have created a separate branch off of `main` in which we can push changes from your fork and can later merge that branch into `main` .

File: nearby-app/src/main/java/com/esri/arcgisruntime/opensourceapps/nearbyplaces/places/PlacesActivity.java [hunk 1]
```diff
@@ -282,34 +259,55 @@ protected void onActivityResult(final int requestCode, final int resultCode, fin
...
+        finish());
     alertDialog.create().show();
   }
-
-  @Override
-  public void onBackPressed() {
-    int count = getSupportFragmentManager().getBackStackEntryCount();
-    if (count == 0) {
-      super.onBackPressed();
+  /*
```
Comment (puneet-pdx) [inline nearby-app/src/main/java/com/esri/arcgisruntime/opensourceapps/nearbyplaces/places/PlacesActivity.java:273]: add missing *

```
/**
   * Prompt user to turn on location tracking and wireless if needed
   */
```

**Comment by puneet-pdx** [comment]
@MadRatSRP do you think you will have some time to address these comments soon ?

**Comment by MadRatSRP** [comment]
> @MadRatSRP do you think you will have some time to address these comments soon ?

Maybe on the next week

**Comment by puneet-pdx** [comment]
Changes merged to `main` https://github.com/Esri/nearby-android/pull/101
Closing this PR

**Pull request closed without merging**